### PR TITLE
gemfile cleanup and refactor

### PR DESCRIPTION
**Pull request by amitsuryavanshi**

### CODE REVIEW

**Comment by amitsuryavanshi** [comment]
Hi @deobald I have some cleanup on gemfile, including neo4j-rake-tasks gem and some refactor. This should not have any functional impact on dependent repositories. Kindly take a look.

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@ def tmp_path( *paths )
 CarrierWave.root = public_path
 # DatabaseCleaner[:neo4j, connection: {type: :bolt, path: 'bolt://localhost:7006'}].strategy = :transaction
 
-neo4j_adaptor = Neo4j::Core::CypherSession::Adaptors::Bolt.new('bolt://localhost:7006', {ssl: false})
+neo4j_adaptor = Neo4j::Core::CypherSession::Adaptors::Bolt.new('bolt://localhost:7472', {ssl: false})
```
Comment (deobald) [inline spec/spec_helper.rb:32]: Is this a standard port for testing databases, of some sort? It's oddly aligned with the HTTP/HTTPS ports, which is why I ask. If, on the other hand, it's just as arbitrary as 7006, I'm not sure I understand why this change is necessary.

File: lib/carrierwave/neo4j.rb [hunk 2]
```diff
@@ -210,4 +208,4 @@ def store_previous_changes_for_#{column}
   end
 end
 
-Neo4j::ActiveNode.send :include, CarrierWave::Neo4j
+Neo4j::ActiveNode.include CarrierWave::Neo4j
```
Comment (deobald) [inline lib/carrierwave/neo4j.rb:211]: You created a sample app and tested this out to make sure it works?

**Review by deobald — COMMENTED** [review]
I've asked a couple questions in-line. The addition of `neo4j-rake_tasks` makes sense for setting up a local environment. If you want to describe how you set up your environment in the README (https://github.com/pariyatti/carrierwave-neo4j/#development), that would definitely be beneficial to the next person who attempts to update this lib -- I can do it, but I'd describe my entire Neo4j.rb setup as haphazard, at best. It would be nice to know gem users (myself included) are doing things the way the maintainers expect, actually. For context, I ran off Neo4j Desktop for months because I was under the impression `rake_tasks` was broken and unmaintained.

File: spec/spec_helper.rb (hunk 1 above)
Comment (amitsuryavanshi) [reply spec/spec_helper.rb:32]: Yes, it is a default port.

File: lib/carrierwave/neo4j.rb (hunk 2 above)
Comment (amitsuryavanshi) [reply lib/carrierwave/neo4j.rb:211]: I just ran tests. Tests don't catch this ?

**Comment by amitsuryavanshi** [comment]
I will add the link to `neo4j-rake-tasks` gem README and some description.

File: lib/carrierwave/neo4j.rb (hunk 2 above)
Comment (deobald) [reply lib/carrierwave/neo4j.rb:211]: No. Remember? I called this out very explicitly:

https://github.com/neo4jrb/carrierwave-neo4j/pull/17#discussion_r419090541

If you want to make meta-programming changes to this (_unbelievably_ fragile) gem, please click-test them by standing up a sample app and testing by hand. This fragility is inherent in the way `carrierwave` was written so there isn't really a way to fix it at the `neo4j.rb` level.

As I also mentioned before: I tried to clean up some of the original meta-programming done by @cheerfulstoic (including this change to `.include`) but I couldn't get in touch with him to confirm why things were structured the way they were. The `carrierwave` upstream codebase is sort of an early-2000s Ruby runtime meta-programming trash fire... I assume that's why Brian needed to do some of the things he did -- and also why the test harness can't faithfully recreate objects in MRI which behave in an identical way to how they behave inside a real Rails app. Because `Neo4j::ActiveNode` is a module instead of an abstract class, it wasn't possible for me to follow the ActiveRecord example here, either.

Sorry about this. I know it's a hassle... the week I spent bringing the gem back to life is a week of my life I'll never get back. I feel your pain. :face_with_head_bandage:

Comment (deobald) [reply lib/carrierwave/neo4j.rb:211]: I just moved our app to your branch and tested this out (as thoroughly as I could think to do, anyway)... it certainly didn't break in any of the obvious ways that my changes to that `send :include` line broke things when I was working on the gem, so hopefully everything is kosher.

The change makes me really nervous and I'm trying to instill that nervousness in you, @amitsuryavanshi  :wink: ...if it turns out that this gem is having weird behaviour in the future, let's please remember to revisit this.

Just for my own sanity, I assume the change linked below is what makes the call to `.include` possible? I'm pretty sure I didn't do this when I tried this change originally and if you know this is what makes it work, the whole thing would freak me out a lot less since we'd be riding on "Amit knows this is the right way to do it" rather than "Steven tried it in an app and it didn't seem to break." :stuck_out_tongue_winking_eye: 

https://github.com/pariyatti/carrierwave-neo4j/pull/1/files/1314f41a4d5501c5833e5a91b9a98afb38e9b3bf#diff-823bbce648661b3ecda19b60e9731b65L106

**Comment by deobald** [comment]
Everything looks good. Once you've added your changes to the README, I can merge this PR.

File: lib/carrierwave/neo4j.rb (hunk 2 above)
Comment (amitsuryavanshi) [reply lib/carrierwave/neo4j.rb:211]: Pardon me, lot of things on my plate, forgot your comment on other PR. I assume we should be safe with these changes.
